woocommerce/woocommerce-gutenberg-products-block
# Refactor useBillingData to useCheckoutContext and hooking it up to form

So this PR changes some things:

1- previously, the returned shipping address didn’t have first and last name in it, this PR adds it.
2- It refactors billingFields to be saved in `useCheckoutContext` reducer, instead of locally.
3- `useBillingData` now exposes six properties:
`email` `setEmail` `billingAddress` `setBillingAddress` `billingData` `setBillingData`.
this is meant to track and update each of them separately, although I think simply `billingData` and `setBillingData` should be enough.
4. It removes the `use shipping as billing` option in the editor since it was useless now.


fixes https://github.com/woocommerce/woocommerce-gutenberg-products-block/issues/1940

### How to test the changes in this Pull Request:

1. Make sure you have some default billing and shipping information in Woo in your profile -> billing
2. in Checkout block, see that those values are hydrated.
3. Update the shipping form
3. toggle off the use shipping as billing option
4. see that billing forms are the same as shipping
5. update shipping now again
6. those changes should not be reflected on billing
7. Toggling and untoggling that option will copy the values.
